http-protocol-binding.md

```diff
@@ -213,16 +213,75 @@ and per [RFC7230, section 3][rfc7230-section-3], HTTP headers MUST only use
 printable characters from the US-ASCII character set, and are terminated by a
 CRLF sequence with OPTIONAL whitespace around the header value.
 
-String values MUST be percent-encoded as described in [RFC3986, section
-2.4][rfc3986-section-2-4] before applying the header encoding rules described in
-[RFC7230, section 3.2.6][rfc7230-section-3-2-6].
-
-When decoding an HTTP message into a CloudEvent, these rules MUST be applied in
-reverse -- [RFC7230, section 3.2.6][rfc7230-section-3-2-6] decoding to an ASCII
-string, and then a **single round** of percent-decoding as described in
-[RFC3986, section 2.4][rfc3986-section-2-4] to produce a valid UTF-8 String.
-(Note that applying percent-decoding an incorrect number of times can result in
-message corruption or security issues.)
+When encoding a CloudEvent as an HTTP message, string values
+represented as HTTP header values MUST be percent-encoded as
+described below. This is compatible with [RFC3986, section
+2.1][rfc3986-section-2-1] but is more specific about what needs
+encoding. The resulting string does not then need any further
```

Comment:
"does not then need" smells like a "MUST NOT"?

Reply:
Not particularly... it would be *okay* to quote things (it wouldn't break anything) but we wouldn't encourage SDKs to do that, as there's no point. (Just like it's *valid* for an SDK to escape every character, even ASCII alphanumeric ones. It's just pointless and wasteful.)

I'd be happy to add a "SHOULD NOT" for this, to make the discouragement clear?

Reply:
"does not then need" dances appears to dance around least one of the ALLCAPS words. If SHOULD NOT is what you mean, that is what you should make it.

Reply:
This is what's in the new commit:

> The resulting string SHOULD NOT be further encoded.
> (Rationale: quoted string escaping is unnecessary when every space
> and double-quote character is already percent-encoded.)